workshop_4/1_recap_of_workshop_3.ipynb

```diff
@@ -0,0 +1,197 @@
+   "metadata": {},
+   "source": [
+    "You see the main steps here. On the first line:\n",
+    "\n",
+    "- The `def` keyword indicates we're making a function\n",
+    "- We give it a name\n",
+    "- We define the input parameters in brackets\n",
+    "- Don't forget the colon on the end of the line!\n",
+    "\n",
+    "Everything on the lines below that is indented is part of the function.\n",
+    "\n",
+    "- First comes the docstring - some information for people using your function to help them understand the code.\n",
+    "- Then comes the code itself\n",
+    "- Then we have the `return` command, which is how we get outputs like numbers, text, data, or anything else out of the function\n",
+    "\n",
+    "This function takes a number as in input and returns the suppressed/rounded number as its output."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Calling a function\n",
+    "\n",
+    "If we want to use this function, here's how we do it:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "suppressed_number = suppress_number(23)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We just invoke the function's name, and pass it an input in brackets. Easy as that!\n",
+    "\n",
+    "Anything that we returned from the function will now be stored in the suppressed_number variable.\n",
+    "\n",
+    "Let's see what happens when we print that:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(suppressed_number)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Importing Files"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Sometimes we want to use some code that's in a different file to the one we're working on. For example, maybe we want to call a function that was defined in a different file.\n",
+    "\n",
+    "To do this, we need to import the file.\n",
+    "\n",
+    "Look at this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from src.processing import aggregations"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The the bit after the `from` is actually a directory path. It's saying from src/processing import a file called aggregations.py\n",
+    "\n",
+    "Once imported, we can call the functions in the file - we just need to let Python know which file we want it to look in. For example if there was a function called `calculate_regional_breakdowns()` in that file, we could call it like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "aggregations.calculate_regional_breakdowns()"
```

Comment:
Is the function import in the recap supposed to work? It could be confusing to people running it and it failing.